torch/utils/data/dataset.py

```diff
@@ -253,6 +254,61 @@ def __len__(self):
         return total
 
 
+class BufferedShuffleDataset(IterableDataset[T_co]):
+    r"""Dataset shuffled from the original dataset.
+
+    This class is useful to shuffle an existing instance of an IterableDataset.
+    The buffer with buffer_size is filled with the items from the dataset first. Then,
+    each item will be yielded from the buffer by reservoir sampling via iterator.
+
+    For `buffer_size` smaller than 1, the `buffer_size` is set to 1 representing
```

Comment:
why do we want to support `buffer_size < 1`?  We could just error out, no?

Reply:
You are right for `buffer_size<1`. But, how about the `buffer_size==1`? It's not actually useful for shuffling.

Reply:
For `buffer_size == 1`, the code should still work, and `BufferedShuffleDataset` is the identity.

Reply:
I see. I did update it with assertion `buffer_size > 0`.